### PR TITLE
Remove deprecated torch.symeig

### PR DESCRIPTION
**Pull request by IvanYashchuk**
The time has come to remove deprecated linear algebra related functions. This PR removes `torch.symeig`.

- [x] XLA PR: https://github.com/pytorch/xla/pull/4498

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @xwang233 @Lezcano @mcarilli @ptrblck @leslie-fang-intel @jgong5 @IvanYashchuk

### CODE REVIEW

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cpp [hunk 1]
```diff
@@ -2337,118 +2337,6 @@ std::tuple<Tensor, Tensor> _linalg_qr_helper_cuda(const Tensor& input, c10::stri
...
-    lrwork = magma_int_cast(std::max<int64_t>(1, rwkopt), "rwork_size");
-    storage_rwork = pin_memory<value_t>(lrwork);
-    rwork = static_cast<value_t*>(storage_rwork.data());
-  }
-
-  for (decltype(batch_size) i = 0; i < batch_size; i++) {
-    scalar_t* vectors_working_ptr = &vectors_data[i * vectors_stride];
-    value_t* values_working_ptr = &values_data[i * values_stride];
-    magma_int_t* info_working_ptr = &infos_data[i];
-    magmaSyevd<scalar_t, value_t>(jobz, uplo, n, vectors_working_ptr, lda, values_working_ptr,
```
Comment (Lezcano) [inline aten/src/ATen/native/cuda/BatchLinearAlgebra.cpp:2404]: We're not using this MAGMA function anymore in `eigh`?

File: tools/autograd/derivatives.yaml [hunk 2]
```diff
@@ -1461,11 +1461,8 @@
 - name: _svd_helper(Tensor self, bool some, bool compute_uv) -> (Tensor U, Tensor S, Tensor V)
   self: svd_backward(grads, self, some, compute_uv, U, S, V)
 
-- name: symeig(Tensor self, bool eigenvectors=False, bool upper=True) -> (Tensor eigenvalues, Tensor eigenvectors)
-  self: eigh_backward(grads, self, eigenvectors, eigenvalues, eigenvectors_return)
-
 - name: linalg_eigh(Tensor self, str UPLO="L") -> (Tensor eigenvalues, Tensor eigenvectors)
-  self: eigh_backward(grads, self, /*eigenvectors=*/true, eigenvalues, eigenvectors)
```
Comment (Lezcano) [inline tools/autograd/derivatives.yaml:1468]: This is going to conflict when you rebase as https://github.com/pytorch/pytorch/pull/70528 was merged.
You'll need to remove the `/*symeig_eigenvector=*/eigenvectors)` argument from the function signature and the logic it encapsulates in the `eig_backward` function itself.

**Review by lezcano — COMMENTED** [review]
Just a couple points

**Comment by mruberry** [comment]
@IvanYashchuk this one looks like it has some build issues

**Comment by IvanYashchuk** [comment]
Thanks! I probably messed up when resolving merge conflicts. I'll fix it now.

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cpp (hunk 1 above)
Comment (IvanYashchuk) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cpp]: It seems that it was removed by mistake, in the current state of the PR this function (`apply_magma_eigh`) remains in the codebase and is used.

**Review by lezcano — APPROVED** [review]
This looks good, thanks!

**Comment by kit1980** [comment]
@IvanYashchuk Can we revive this? I think I've updated all important Meta-internal usages of `torch.symeig`.

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
@IvanYashchuk can you resolve conflicts?
Other than that, should be good to go - I didn't see any internal issues.

**Comment by kit1980** [comment]
@IvanYashchuk can you forward-fix XLA tests?

AssertionError: Found an invalid operator name: symeig
https://github.com/pytorch/pytorch/actions/runs/3992628737/jobs/6848818658

**Comment by IvanYashchuk** [comment]
Yes, I'll do that today.

**Comment by kit1980** [comment]
The XLA fix is causing problems, I'm reverting this for now so we have passing XLA and don't miss more issues.

**Review by malfet — APPROVED** [review]

**Comment by IvanYashchuk** [comment]
XLA is passing: https://github.com/pytorch/pytorch/actions/runs/4046657961/jobs/6960036431

**Comment by IvanYashchuk** [comment]
Some CI jobs are failing (e.g. https://github.com/pytorch/pytorch/actions/runs/4048928070/jobs/6974815885):
```
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-py3.8-gcc7:82d9d2b1e97524ec4988a070c31f4a2655ef32a3 not found: manifest unknown: Requested image not found
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-py3.8-gcc7:82d9d2b1e97524ec4988a070c31f4a2655ef32a3 not found: manifest unknown: Requested image not found
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-py3.8-gcc7:82d9d2b1e97524ec4988a070c31f4a2655ef32a3 not found: manifest unknown: Requested image not found
Error: Process completed with exit code 1.
```

**Pull request closed without merging**